pyo3-macros-backend/src/attrs.rs

```diff
@@ -0,0 +1,27 @@
+use proc_macro2::TokenStream;
+use quote::ToTokens;
+use syn::spanned::Spanned;
+use syn::{Meta, MetaNameValue, Result};
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct FromPyWithAttribute(syn::ExprPath);
+
+impl FromPyWithAttribute {
+    pub fn from_meta(meta: Meta) -> Result<Self> {
+        if let Meta::NameValue(MetaNameValue {
+            lit: syn::Lit::Str(string_literal),
+            ..
+        }) = meta
```

Comment:
I personally prefer do something like

```rust
let string_literal = match meta {
    Meta::NameValue(MetaNameValue { lit: syn::Lit::Str(string_literal), .. } => string_literal,
    meta => bail_spanned!(meta.span() => "expected a name-value: `pyo3(from_py_with = \"func\")` "),
};
```

... so that the rest of the function isn't indented one level.

But I'm happy as-is if you prefer the current form.

Reply:
Thanks! I agree, less indentation is better